openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -0,0 +1,250 @@
+                 consecutive requests via the `get` method
+    batch_limit: integer giving the number of records to get in each batch
+    retries:     integer number of times to retry the request on error
+    headers:     dictionary to be passed as headers to the request
+    """
+
+    delay = 1
+    retries = 3
+    batch_limit = 100
+    headers: Dict = {}
+
+    @property
+    @abstractmethod
+    def providers(self) -> dict[str, str]:
+        """
+        A dictionary whose keys are the supported `media_types`, and values are
+        the `provider` string in the `media` table of the DB for that type.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def endpoint(self):
+        """
+        The URL with which to request records from the API.
+        """
+        pass
+
+    def __init__(self, date: str = None):
+        """
+        Optional Arguments:
+        date: Date String in the form YYYY-MM-DD. This is the date for
+              which running the script will pull data
+        """
+        # An airflow variable used to cap the amount of records to be ingested.
+        # This can be used for testing purposes to ensure a provider script
+        # processes some data but still returns quickly.
+        # When set to 0, no limit is imposed.
+        self.limit = Variable.get(
+            "ingestion_limit", deserialize_json=True, default_var=0
+        )
+
+        # If a test limit is imposed, ensure that the `batch_limit` does not
+        # exceed this.
+        if self.limit:
+            self.batch_limit = min(self.batch_limit, self.limit)
+
+        # Initialize the DelayedRequester and all necessary Media Stores.
+        self.delayed_requester = DelayedRequester(self.delay)
+        self.media_stores = self.init_media_stores()
+        self.date = date
+
+    def init_media_stores(self) -> dict[str, MediaStore]:
+        """
+        Initialize a media store for each media type supported by this
+        provider.
+        """
+        media_stores = {}
+
+        for media_type, provider in self.providers.items():
+            StoreClass = get_media_store_class(media_type)
+            media_stores[media_type] = StoreClass(provider)
+
+        return media_stores
+
+    def ingest_records(self, **kwargs) -> None:
+        """
+        The main ingestion function that is called during the `pull_data` task.
+
+        Optional Arguments:
+        **kwargs: Optional arguments to be passed to `get_next_query_params`.
+        """
+        should_continue = True
+        record_count = 0
+        query_params = None
+
+        logger.info(f"Begin ingestion for {self.__class__.__name__}")
+
+        while should_continue:
+            query_params = self.get_next_query_params(query_params, **kwargs)
+
+            batch, should_continue = self.get_batch(query_params)
+
+            if batch and len(batch) > 0:
+                record_count = self.process_batch(batch)
```

Comment:
```suggestion
                record_count += self.process_batch(batch)
```

Reply:
I'm not sure if this is the best coding style for this, but currently we are resetting the record_count with each batch.